src/util/redis/redisConnector.js

```diff
@@ -0,0 +1,158 @@
+  setTimeout(
+    () => reject(new Error("Timeout while connecting to redis")),
+    1000
+  );
+});
+
+const RedisDB = {
+  init() {
+    if (process.env.USE_REDIS_DB && process.env.USE_REDIS_DB !== 'false') {
+      this.host = process.env.REDIS_HOST || 'localhost';
+      this.port = parseInt(process.env.REDIS_PORT, 10) || 6379;
+      this.password = process.env.REDIS_PASSWORD;
+      this.userName = process.env.REDIS_USERNAME;
+      this.maxRetries = parseInt(process.env.REDIS_MAX_RETRIES, 10) || 5;
+      this.timeAfterRetry = parseInt(process.env.REDIS_TIME_AFTER_RETRY_IN_MS, 10) || 500;
+      this.client = new Redis({
+        host: this.host,
+        port: this.port,
+        password: this.password,
+        username: this.userName,
+        enableReadyCheck: true,
+        retryStrategy: (times) => {
+          if (times <= this.maxRetries) {
+            return (1 + times) * this.timeAfterRetry; // reconnect after 
+          }
+          stats.increment("redis_error", {
+            operation: 'redis_down',
+          });
+          log.error(`Redis is down at ${this.host}:${this.port}`);
+          return false; // stop retrying
+        },
+      });
+      this.client.on('ready', () => {
+        log.info(`Connected to redis at ${this.host}:${this.port}`);
+      });
+    }
+  },
+
+
+  async checkRedisConnectionReadyState() {
+    try {
+      await this.client.connect();
+    } catch (error) {
+      return new Promise((resolve) => {
+        this.client.on('ready', () => {
+          resolve();
+        });
+      });
+    }
+    return Promise.resolve();
+  },
+
+  /**
+   * Checks connection with redis and if not connected, tries to connect and throws error if connection request fails
+   */
+  async checkAndConnectConnection() {
+    if (!this.client || this.client.status === "end") {
+      this.init();
+    }
+    if (this.client.status !== 'ready') {
+      await Promise.race([this.checkRedisConnectionReadyState(), timeoutPromise()]);
+    }
+  },
+  /**
+   * Used to get value from redis depending on the key and the expected value type
+   * @param {*} hashKey parent key 
+   * @param {*} isObjExpected false if fetched value can not be json
+   * @param {*} key  key for which value needs to be extracted, required if isObjExpected is true
+   * @returns value which can be json or string or number
+   * storage of data in case isObjExpected is true
+   * hashKey:{
+   * key1: {internalKey1:val1},
+   * key2: {internalKey2:val2},
+   * }
+   */
+  async getVal(hashKey, key, isObjExpected = true) {
+    try {
+      await this.checkAndConnectConnection(); // check if redis is connected and if not, connect
+      let value;
+      let redisVal;
+      if (isObjExpected === true) {
+        redisVal = await this.client.hmget(hashKey, key);
+        if (isDefinedAndNotNull(redisVal)) {
+          value = redisVal.map(element => {
+            try {
+              return JSON.parse(element);
```

Comment:
[Opt.] Can we use `isObject` in our `v0/util/index.js` ? Instead having a try-catch ?!

Also in this function would we also get array type results ?!

Reply:
Got it 👍🏼